[Bug] Headscale crashes when derp.update_interval is not set

### Is this a support request?

- [x] This is not a support request

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

Headscale seems to crash when `derp.urls` is set but not `update_frequency`.

It seems to load a `0` as the default for update interval, which seems to lead to a panic.

With headscale 0.23.0 this worked fine.

```
panic: non-positive interval for NewTicker
/home/runner/work/headscale/headscale/hscontrol/app.go:616 +0xa65
created by github.com/juanfont/headscale/hscontrol.(*Headscale).Serve in goroutine 1
/home/runner/work/headscale/headscale/hscontrol/app.go:248 +0xd4
github.com/juanfont/headscale/hscontrol.(*Headscale).scheduledTasks(0xc00065f4a0, {0x2548310, 0xc0003fe370})
/nix/store/jfv85qbj4vb1dafcg6kncg4vrbq2bbxv-go-1.23.4/share/go/src/time/tick.go:38 +0xbb
time.NewTicker(0x12a05f200?)
```

### Expected Behavior

Run with default `update_frequency`

### Steps To Reproduce

Run headscale with this config:

```
    server_url: https://*****
    listen_addr: 0.0.0.0:8080
    metrics_listen_addr: 0.0.0.0:9090

    noise:
      private_key_path: /var/lib/headscale/noise_private.key

    prefixes:
      allocation: random

      v4: 100.64.0.0/10
      v6: fd7a:115c:a1e0:f00::/64

    database:
      type: sqlite
      sqlite:
        path: /var/lib/headscale/db.sqlite

    derp:
      urls:
        - https://controlplane.tailscale.com/derpmap/default

    dns:
      base_domain: int.vpn
```

### Environment

```markdown
Official 0.24.0 Container
```

### Runtime environment

- [x] Headscale is behind a (reverse) proxy
- [x] Headscale runs in a container

### Anything else?

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,8 @@
   [#2364](https://github.com/juanfont/headscale/pull/2364)
 - Remove invalid routes and add stronger constraints for routes to avoid API panic
   [#2371](https://github.com/juanfont/headscale/pull/2371)
+- Fix panic when `derp.update_frequency` is 0
+  [#2368](https://github.com/juanfont/headscale/pull/2368)
 
 ## 0.24.0 (2025-01-17)
 

--- a/hscontrol/app.go
+++ b/hscontrol/app.go
@@ -245,11 +245,11 @@ func (h *Headscale) scheduledTasks(ctx context.Context) {
 
 	lastExpiryCheck := time.Unix(0, 0)
 
-	derpTicker := time.NewTicker(h.cfg.DERP.UpdateFrequency)
-	defer derpTicker.Stop()
-	// If we dont want auto update, just stop the ticker
-	if !h.cfg.DERP.AutoUpdate {
-		derpTicker.Stop()
+	derpTickerChan := make(<-chan time.Time)
+	if h.cfg.DERP.AutoUpdate && h.cfg.DERP.UpdateFrequency != 0 {
+		derpTicker := time.NewTicker(h.cfg.DERP.UpdateFrequency)
+		defer derpTicker.Stop()
+		derpTickerChan = derpTicker.C
 	}
 
 	var extraRecordsUpdate <-chan []tailcfg.DNSRecord
@@ -285,7 +285,7 @@ func (h *Headscale) scheduledTasks(ctx context.Context) {
 				h.nodeNotifier.NotifyAll(ctx, update)
 			}
 
-		case <-derpTicker.C:
+		case <-derpTickerChan:
 			log.Info().Msg("Fetching DERPMap updates")
 			h.DERPMap = derp.GetDERPMap(h.cfg.DERP)
 			if h.cfg.DERP.ServerEnabled && h.cfg.DERP.AutomaticallyAddEmbeddedDerpRegion {

